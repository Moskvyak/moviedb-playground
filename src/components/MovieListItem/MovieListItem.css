.movielistitem__image {
    border-radius: 10px;
    width: 100%;
}
.movielistitem__title, .movielistitem__date  {
    font-size: 0.75em;
    text-align: left;
    margin-top: 5px;
}

.movielistitem__rating-wrapper {
    position: absolute;
    top: 5px;
    left: 5px;
}