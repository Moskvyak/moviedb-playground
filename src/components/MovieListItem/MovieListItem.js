import React from 'react';

import RatingItem from '../RatingItem';
import { Link } from 'react-router-dom';

import './MovieListItem.css';

const MovieListItem = React.memo(({ poster_path, id, title, vote_average, release_date }) => {
  const fileSize = `/w200`;
  const posterUrl = `${process.env
    .REACT_APP_ASSETS_URL}${fileSize}${poster_path}`;
  const date = new Date(release_date).toLocaleString('en-us', {
    month: 'long',
    year: 'numeric'
  });
  return (
    <div className="grid-item">
      <Link to={`/${id}`}>
        <img className="movielistitem__image" src={posterUrl} />
      </Link>
      <div className="movielistitem__title color-white">
        {title}
      </div>
      <div className="movielistitem__date color-lightgray">
        {date}
      </div>
      <div className="movielistitem__rating-wrapper">
        <RatingItem rating={vote_average} />
      </div>
    </div>
  );
});

export default MovieListItem;
