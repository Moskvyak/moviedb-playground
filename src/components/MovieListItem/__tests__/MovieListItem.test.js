import React from 'react';
import renderer from 'react-test-renderer';
import MovieListItem from '../MovieListItem';
import { BrowserRouter as Router } from 'react-router-dom';

describe('<MovieListItem />', () => {
  const movie = {
    poster_path: 'testpath',
    id: 'test id',
    title: 'test title',
    vote_average: 7.5,
    release_date: '24/10/2018'
  };

  it('should render with required props', () => {
    const component = renderer.create(
      <Router>
        <MovieListItem {...movie} />
      </Router>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
