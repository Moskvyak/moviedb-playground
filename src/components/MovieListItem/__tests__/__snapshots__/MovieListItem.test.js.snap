// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieListItem /> should render with required props 1`] = `
<div
  className="grid-item"
>
  <a
    href="/test id"
    onClick={[Function]}
  >
    <img
      className="movielistitem__image"
      src="undefined/w200testpath"
    />
  </a>
  <div
    className="movielistitem__title color-white"
  >
    test title
  </div>
  <div
    className="movielistitem__date color-lightgray"
  >
    Invalid Date
  </div>
  <div
    className="movielistitem__rating-wrapper"
  >
    <div
      className="primary-background-color ratingitem"
    >
      75%
    </div>
  </div>
</div>
`;
